compiler/rustc_typeck/src/collect.rs

```diff
@@ -1194,9 +1194,11 @@ fn super_predicates_that_define_assoc_type(
 fn trait_def(tcx: TyCtxt<'_>, def_id: DefId) -> ty::TraitDef {
     let item = tcx.hir().expect_item(def_id.expect_local());
 
-    let (is_auto, unsafety) = match item.kind {
-        hir::ItemKind::Trait(is_auto, unsafety, ..) => (is_auto == hir::IsAuto::Yes, unsafety),
-        hir::ItemKind::TraitAlias(..) => (false, hir::Unsafety::Normal),
+    let (is_auto, unsafety, items) = match item.kind {
+        hir::ItemKind::Trait(is_auto, unsafety, .., items) => {
+            (is_auto == hir::IsAuto::Yes, unsafety, items)
+        }
+        hir::ItemKind::TraitAlias(..) => (false, hir::Unsafety::Normal, &[][..]),
```

Comment:
I think this can just be `&[]` instead of `&[][..]`

Reply:
![screenshot of a compilation error that happens without [..] - in short, expected &[TraitItemRef], found &[_; 0]](https://user-images.githubusercontent.com/38225716/148676081-fefd6a91-def5-4be0-8025-9dd499459845.png)

Reply:
Very disappointing; minimal repro: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=875b8d707eb70e293fc06d51eb6f26e0 — a type annotation at line 1201 does fix it, but without that a "wide pointer" _field_ on one branch is not sufficient for a coercion to happen on other branches.

Reply:
Actually, even a one element tuple is enough to demonstrate this: [(play)](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=7f24ff6da9372f4f067c54b8133db21b)